modules/digitaluniverse/rendering/renderableplanescloud.cpp

```diff
@@ -1099,12 +1099,43 @@ void RenderablePlanesCloud::createPlanes() {
     if (_dataIsDirty && _hasSpeckFile) {
         LDEBUG("Creating planes...");
         float maxSize = 0.f;
+        double maxRadius = 0.0;
         for (size_t p = 0; p < _fullData.size(); p += _nValuesPerAstronomicalObject) {
+            float scale = 0.f;
+            switch (_unit) {
+                case Meter:
+                    scale = 1.f;
+                    break;
+                case Kilometer:
+                    scale = 1e3f;
+                    break;
+                case Parsec:
+                    scale = static_cast<float>(PARSEC);
+                    break;
+                case Kiloparsec:
+                    scale = static_cast<float>(1e3 * PARSEC);
+                    break;
+                case Megaparsec:
+                    scale = static_cast<float>(1e6 * PARSEC);
+                    break;
+                case Gigaparsec:
+                    scale = static_cast<float>(1e9 * PARSEC);
+                    break;
+                case GigalightYears:
+                    scale = static_cast<float>(306391534.73091 * PARSEC);
+                    break;
+            }
+
```

Comment:
I like the switch case format above in RenderableBillboardsCloud::unitToMeter. How about doing it that way leaving scale as a double, and then casting to a float after the switch case?

Reply:
Good point  -> changed